vortex-array/src/array/list/mod.rs

```diff
@@ -142,6 +145,32 @@ impl ListArray {
             .child(0, dtype, self.metadata().elements_len)
             .vortex_expect("array contains elements")
     }
+
+    // This is a convenience method to create a list array from an iterator of iterators.
+    // This method is slow however since each element is first converted to a scalar and then
+    // appended to the array.
+    pub fn from_iter_slow<I: IntoIterator>(iter: I, dtype: Arc<DType>) -> VortexResult<ArrayData>
```

Comment:
1. Comment should be rustdoc to make it clear that it shouldn't be used.
2. IDK if we want this as part of the public API, maybe it should be in some `test-utils` module or something to make it harder to misuse.

Reply:
Done 1, is 2 a pattern elsewhere?

Reply:
Do you think it should be a top level array module or nested in list?

Reply:
I know some create feature flag/have a tests module for stuff like that (for example - arrow-rs has a `test_utils` feature)

Reply:
Then it cannot be a method on `ListArray`. Maybe we can just use a feature flag and a nested module?

Reply:
why can't it be a method on `ListArray` with a feature flag?

Reply:
It can in but only if its in the same module (i.e. not in `test_utils`)

Reply:
we can just stick with a feature flag, should be enough to keep it out of the default-available public API